imports/client/components/ChatPeople.tsx

```diff
@@ -0,0 +1,468 @@
+      const nextState = {
+        deafened: !prevState.deafened,
+      };
+
+      const prevEffectiveMuteState = prevState.muted || prevState.deafened;
+      const nextEffectiveMuteState = prevState.muted || nextState.deafened;
+
+      // Update gain if needed
+      if (prevEffectiveMuteState !== nextEffectiveMuteState) {
+        const newGain = nextEffectiveMuteState ? 0.0 : 1.0;
+        if (prevState.gainNode && prevState.audioContext) {
+          prevState.gainNode.gain.setValueAtTime(newGain, prevState.audioContext.currentTime);
+        }
+      }
+
+      const effectiveState = participantState(prevState.muted, nextState.deafened);
+      if (props.selfParticipant) {
+        Meteor.call('setCallParticipantState', props.selfParticipant._id, effectiveState,
+          (err: Meteor.Error | undefined) => {
+            if (err) {
+              // Ignore.  Not much we can do here; the server failed to accept our change.
+            }
+          });
+      }
+
+      return nextState;
+    });
+  };
+
+  joinCall = () => {
+    if (navigator.mediaDevices) {
+      this.setState({
+        state: CallState.REQUESTING_STREAM,
+      });
+
+      // Get the user media stream.
+      const mediaStreamConstraints = {
+        audio: {
+          echoCancellation: { ideal: true },
+          autoGainControl: { ideal: true },
+          noiseSuppression: { ideal: true },
+        },
+        // TODO: conditionally allow video if enabled by feature flag?
+      };
+
+      navigator.mediaDevices.getUserMedia(mediaStreamConstraints)
+        .then(this.gotLocalMediaStream)
+        .catch(this.handleLocalMediaStreamError);
+    } else {
+      this.setState({
+        state: CallState.STREAM_ERROR,
+        error: 'Couldn\'t get local microphone: browser denies access on non-HTTPS origins',
+      });
+    }
+  };
+
+  gotLocalMediaStream = (mediaStream: MediaStream) => {
+    // @ts-ignore ts doesn't know about the possible existence of webkitAudioContext
+    const AudioContext = window.AudioContext || window.webkitAudioContext;
+    const audioContext = new AudioContext();
+    const wrapperStreamDestination = audioContext.createMediaStreamDestination();
+    const gainNode = audioContext.createGain();
+    const gainValue = this.state.muted ? 0.0 : 1.0;
+    gainNode.gain.setValueAtTime(gainValue, audioContext.currentTime);
+
+    const leveledStreamSource = new MediaStream();
+    const rawTracks = mediaStream.getTracks();
+    for (let i = 0; i < rawTracks.length; i++) {
+      const rawTrack = rawTracks[i];
+      if (rawTrack.kind === 'audio') {
+        // Chrome doesn't support createMediaStreamTrackSource, so stuff the
+        // track in another stream.
+        const stubStream = new MediaStream();
+        stubStream.addTrack(rawTrack);
+        const wrapperStreamSource = audioContext.createMediaStreamSource(stubStream);
+
+        // Wire up the audio track to the gain node.
+        wrapperStreamSource.connect(gainNode);
```

Comment:
I believe in general `getUserMedia` is expected to return at most one audio track and at most one video track (and we specify that we only want audio for now).  I'm still mulling the thought of if I want to make video possible in the future, potentially to support screensharing (probably more for smaller group events than for the MIT hunt in particular).

You have a reasonable point about that being a future requirement though :)

Reply:
Yeah. I'm pretty sure this always works as written. I just find the structure strange because it suggests iteration that should never happen (and wouldn't work very sensibly if it did). But you're much more likely to be maintaining this than I am, so I leave the style decision to you!

Reply:
As a potential future cleanup, maybe `const audioTrack = rawTracks.find(t => t.kind === 'audio')`?